wpcom-legacy-redirector.php

```diff
@@ -86,7 +87,7 @@ static function insert_legacy_redirect( $from_url, $redirect_to ) {
 
 		$from_url_hash = self::get_url_hash( $from_url );
 
-		if ( false !== self::get_redirect_uri( $from_url ) ) {
+		if ( false !== self::get_redirect_uri( $from_url ) && ( 'add' === $mode ) ) {
```

Comment:
Instead of checking for a duplicate via `get_redirect_uri`, we can use `get_redirect_post_id` instead and then don't have to do the lookup further down.

```
$redirect_id = self::get_redirect_post_id( $from_url, false );
if ( $redirect_id && ! $update_if_exists ) {
    return new WP_Error( 'duplicate-redirect-uri', 'A redirect for this URI already exists' );
}
```